### ISSUE TITLE
OAK-FFC-4P: 4-Camera FFC USB3C DepthAI Variant

### ISSUE THREAD

**Issue by Luxonis-Brandon** [luxonis/depthai-hardware#85]
### Start with the `why`:

For the 1099 SOM, we have the 1090FFC board, which allows folks to quickly prototype with various cameras (e.g. from ArduCam [here](https://docs.luxonis.com/projects/hardware/en/latest/pages/arducam.html)), stereo baselines, filters (for hyperspectral systems), etc. quickly.  We are currently lacking such a board for the 2099 SOM.

The 2099 SOM supports several additional things that the 1099 SOM does not, including:

1. 4 cameras (instead of 3): 2x 4-lane, 2x 2-lane
2. uSD Support

So we need a board that allows such prototyping, while also enabling the additional functionality of the 2099 SOM above.

On the 1090FFC board, we tried to make it as small as possible so that when folks prototype, their overall prototype is not huge (as we have found that most of our customers need to use even their prototypes in space-constrained environments).  We have actually found that even though we tried to make it small, it is still borderline "too big" for many customers, or outright is too big.

So for the 2090FFC variant, it is key to try to keep it as small, or close to as small, as the 1090FFC.  

As this 4-camera board may be used with a variety of cameras, including those with hardware frame-sync, there should be a system onboard for allowing up to all 4 cameras to be synchronized.

### Move to the `how`:

Increase layers as necessary from the 1090FFC, and spend additional routing/placement time/effort to make the 2090FFC as compact as reasonably possible.  And we can also change/remove connectors/etc. to do so.  

Likely use slots for the mounting holes, like below, to maximize the usable board area while minimizing the actual size of the device:
![image](https://user-images.githubusercontent.com/32992551/124319629-bb7abf00-db37-11eb-80b7-b9a419d4350b.png)

### Move to the `what`:

A compact, 4-camera-capable FFC board with onboard SD-Card and IMU with onboard hardware allowing all 4 cameras to be hardware synced (FSIN).

**Comment by Luxonis-Brandon** [comment]
Here's what we're thinking:

![image](https://user-images.githubusercontent.com/32992551/125143123-d4a2e300-e0d6-11eb-9e34-591d37245596.png)
![image](https://user-images.githubusercontent.com/32992551/125143127-d8366a00-e0d6-11eb-9edf-efb9e5746c00.png)


The PRBS connector is new to this design, meaning Power, Reset, Boot, and Sync.  As those allude, that single connector allows provideing the following:

1. Power (instead of over USB or barrel)
2. Reset (for remote hardware reset of DepthAI, which can be useful for power savings, allowing a host to hold DepthAI in reset)
3. Boot mode.  This allows controlling DepthAI boot mode remotely.  The IO line can control between NOR flash boot and USB boot.  This line is used in conjunction with reset to be able to for example flash the NOR flash over USB.
4. Sync.  This exposes the onboard sync (FISN) between the 4 FFC cameras.  This would allow externally driving a sync to all 4, or setting one of the 4 as a driver on this board, then outputting to another, or multiple other 2090FFC to have all the cameras by hardware synced.

**Comment by Luxonis-Brandon** [comment]
The PCBAs just finished.  
![image](https://user-images.githubusercontent.com/32992551/131550557-dafd588e-2a3d-4039-b63f-08d96bb50cad.png)
![image](https://user-images.githubusercontent.com/32992551/131550578-f6c409e9-9bfd-4f0e-866f-9596b1e9b799.png)

**Comment by Luxonis-Brandon** [comment]
Working:
![image](https://user-images.githubusercontent.com/32992551/132878681-29569281-54c4-4841-b09d-d4f61974e02f.png)
![imageadsfasgasdfasdflk](https://user-images.githubusercontent.com/32992551/132878968-dab1c8e6-88c4-467b-bebf-4f5fcf2828c9.jpg)

We got the mounts for the heatsink wrong though.  So they are removed from the board for now (hence the little Pi heatsink).  

We've updated our SOM library as ambiguity there is what lead to the mistake.  Fortunately this hasn't happened on any customer designs.
